### PR TITLE
JSDK-2953 - Video Track TS Definitions

### PR DESCRIPTION
**Pull request by PikaJoyce**
This is a PR for the video track TS definitions.

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -18,3 +18,4 @@ yarn.lock
 .idea
 .vscode
 pre-commit.sh
+tsdef/twilio-video-tests.js
```
Comment (mhuynh5757) [inline .gitignore:21]: This shouldn't be necessary if the JS isn't emitted by using `--noEmit` with `tsc`.

File: tsdef/LocalVideoTrack.d.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  id: Track.ID;
+  isStopped: boolean;
+
+  disable(): LocalVideoTrack;
+  enable(enabled?: boolean): LocalVideoTrack;
+  restart(constraints?: MediaTrackConstraints): Promise<void>;
+  stop(): LocalVideoTrack;
+
+  on(event: 'dimensionsChanged', listener: (track: VideoTrack) => void): this;
```
Comment (mhuynh5757) [inline tsdef/LocalVideoTrack.d.ts:16]: Event listeners should be able to return `any`, not just `void`.

File: tsdef/VideoTrack.d.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+export namespace VideoTrack {
+  interface Dimensions {
+      width: number | null;
+      height: number | null;
+  }
+}
+
+export class VideoTrack extends Track {
+  isStarted: boolean;
+  isEnabled: boolean;
```
Comment (manjeshbhargav) [inline tsdef/VideoTrack.d.ts:12]: @PikaJoyce This should be in `Track.d.ts`.
```suggestion
```

File: tsdef/LocalVideoTrack.d.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+  id: Track.ID;
+  isStopped: boolean;
+
+  disable(): LocalVideoTrack;
+  enable(enabled?: boolean): LocalVideoTrack;
+  restart(constraints?: MediaTrackConstraints): Promise<void>;
+  stop(): LocalVideoTrack;
+
+  on(event: 'dimensionsChanged', listener: (track: VideoTrack) => any): this;
```
Comment (manjeshbhargav) [inline tsdef/LocalVideoTrack.d.ts:16]: @PikaJoyce Please change return type of all event listeners to `void` (even the ones already committed).
```suggestion
  on(event: 'dimensionsChanged', listener: (track: VideoTrack) => void): this;
```

File: tsdef/RemoteVideoTrack.d.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import { Track } from './Track';
+import { VideoTrack } from './VideoTrack';
+
+export class RemoteVideoTrack extends VideoTrack {
+  sid: Track.SID;
+  priority: Track.Priority;
```
Comment (manjeshbhargav) [inline tsdef/RemoteVideoTrack.d.ts:6]: @PikaJoyce 
```suggestion
  priority: Track.Priority | null;
```

File: tsdef/RemoteVideoTrack.d.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  isSwitchedOff: boolean;
+  isEnabled: boolean;
+
+  setPriority(priority: Track.Priority): RemoteVideoTrack;
```
Comment (manjeshbhargav) [inline tsdef/RemoteVideoTrack.d.ts:10]: @PikaJoyce 
```suggestion
  setPriority(priority: Track.Priority | null): this;
```

File: tsdef/VideoTrack.d.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+  dimensions: VideoTrack.Dimensions;
+  kind: 'video';
+  mediaStreamTrack: MediaStreamTrack;
+
+  _attachments?: HTMLMediaElement[];
```
Comment (manjeshbhargav) [inline tsdef/VideoTrack.d.ts:17]: @PikaJoyce Please do not add types for private members/methods.
```suggestion
```

File: tsdef/VideoTrack.d.ts [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+  attach(element?: HTMLMediaElement | string): HTMLVideoElement;
+  detach(element?: HTMLMediaElement | string): HTMLMediaElement[];
```
Comment (manjeshbhargav) [inline tsdef/VideoTrack.d.ts:20]: @PikaJoyce 
```suggestion
  detach(element?: HTMLMediaElement | string): HTMLVideoElement[];
```

File: tsdef/LocalVideoTrack.d.ts [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+import { Track } from './Track';
+import { VideoTrack } from './VideoTrack';
+
+export class LocalVideoTrack extends VideoTrack {
+  constructor(mediaStreamTrack: MediaStreamTrack, options?: LocalTrackOptions);
+
+  id: Track.ID;
+  isStopped: boolean;
+
+  disable(): LocalVideoTrack;
```
Comment (manjeshbhargav) [inline tsdef/LocalVideoTrack.d.ts:11]: @PikaJoyce 
```suggestion
  disable(): this;
```

File: tsdef/LocalVideoTrack.d.ts [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+  enable(enabled?: boolean): LocalVideoTrack;
```
Comment (manjeshbhargav) [inline tsdef/LocalVideoTrack.d.ts:12]: @PikaJoyce 
```suggestion
  enable(enabled?: boolean): this;
```

File: tsdef/LocalVideoTrack.d.ts [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+  restart(constraints?: MediaTrackConstraints): Promise<void>;
+  stop(): LocalVideoTrack;
```
Comment (manjeshbhargav) [inline tsdef/LocalVideoTrack.d.ts:14]: @PikaJoyce 
```suggestion
  stop(): this;
```

**Review by manjeshbhargav — CHANGES_REQUESTED** [review]

File: tsdef/RemoteVideoTrack.d.ts [hunk 12]
```diff
@@ -0,0 +1,19 @@
+import { Track } from './Track';
+import { VideoTrack } from './VideoTrack';
+
+export class RemoteVideoTrack extends VideoTrack {
+  sid: Track.SID;
+  priority: Track.Priority | null;
+  isSwitchedOff: boolean;
+  isEnabled: boolean;
+
+  setPriority(priority: Track.Priority): this;
```
Comment (manjeshbhargav) [inline tsdef/RemoteVideoTrack.d.ts:10]: @PikaJoyce 
```suggestion
  setPriority(priority: Track.Priority | null): this;
```

File: tsdef/VideoTrack.d.ts [hunk 13]
```diff
@@ -0,0 +1,22 @@
+import { Track } from './Track';
+
+export namespace VideoTrack {
+  interface Dimensions {
+      width: number | null;
+      height: number | null;
```
Comment (manjeshbhargav) [inline tsdef/VideoTrack.d.ts:6]: @PikaJoyce Nitpick:
```suggestion
    width: number | null;
    height: number | null;
```

**Review by manjeshbhargav — APPROVED** [review]

**Pull request merged**